### PR TITLE
bpo-33164: Blake2 dependency update

### PR DESCRIPTION
**Pull request by ghost**
Updating the module following-up recent changes

https://github.com/BLAKE2/libb2/commit/620681a3b15c4d7239b9323b9da5ea208a959d3d

https://bugs.python.org/issue33164

### CODE REVIEW

**Comment by dcarlier-afilias** [comment]
Had been fixed the windows build issue (had been pushed upstream too).

File: Modules/_blake2/impl/blake2-impl.h [hunk 1]
```diff
@@ -131,8 +135,21 @@ BLAKE2_LOCAL_INLINE(uint64_t) rotr64( const uint64_t w, const unsigned c )
...
-  memset_v(v, 0, n);
+#if defined(_WIN32) || defined(WIN32)
+  SecureZeroMemory(v, n);
+#else
+// prioritize first the general C11 call
+#if defined(HAVE_MEMSET_S)
+  memset_s(v, n, 0, n);
+#elif defined(HAVE_EXPLICIT_BZERO)
+  explicit_bzero(v, n);
+#elif defined(HAVE_EXPLICIT_MEMSET)
```
Comment (methane) [inline Modules/_blake2/impl/blake2-impl.h:149]: Where these `HAVE_*` macros come from?

Comment (ghost) [reply Modules/_blake2/impl/blake2-impl.h:149]: Good point. I regenerated configure.ac/configure in the last commit.

**Comment by methane** [comment]
Please revert `aclocal.m4` unless it's mandatory.

**Comment by methane** [comment]
I want more verbose NEWS entry (e.g. "Updated blake2 implementation which uses secure memset implementation provided by platform.")

**Comment by dcarlier-afilias** [comment]
Any update ? :)

**Comment by tiran** [comment]
One builder is failing. Please rebase your PR and push again. This should trigger a rebuild.

**Review by tiran — CHANGES_REQUESTED** [review]
The code in Modules/_blake2/impl is a copy of the reference implementation from https://github.com/BLAKE2. Please get your patch into the reference implementation first.

**Comment by dcarlier-afilias** [comment]
Module code updated.

**Comment by csabella** [comment]
@dcarlier-afilias, please rebase to fix merge conflicts.  Thanks!

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by dcarlier-afilias** [comment]
Any chance it lands for 3.8 or will it be 3.9 (at best) ? thx.

**Comment by csabella** [comment]
@dcarlier-afilias, this is a massive change, so my guess is that it would be difficult to land it in 3.8.  I had flagged @tiran and @methane for review and asked for the rebase just to begin the process.  

> The code in Modules/_blake2/impl is a copy of the reference implementation from https://github.com/BLAKE2. Please get your patch into the reference implementation first.

I assume you took care of @tiran's original review comment?  My guess is that you did based on your followup comment, but I just want to make sure.

**Comment by dcarlier-afilias** [comment]
I pushed upstream what [I deemed necessary](https://github.com/BLAKE2/libb2/commits/master?author=dcarlier-afilias) within a year indeed. I hope it is ok.

**Comment by csabella** [comment]
@dcarlier-afilias, looks like I was wrong about it landing in 3.8.  🙂  Thank you for the contribution!  @methane, thank you for the review and the merge!  :tada: 🎉

**Pull request merged**